File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    position = {name: i for i, name in enumerate(players)}

    for name in callings:
        curr_pos = position[name]
        players[curr_pos], players[curr_pos - 1] = players[curr_pos - 1], players[curr_pos]

        position[players[curr_pos]] = curr_pos
        position[name] = curr_pos - 1

    return players

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    let position = {};
    players.forEach((player, index) => {
        position[player] = index;
    });
    
    callings.forEach(calling => {
        let currIndex = position[calling];
        [players[currIndex], players[currIndex - 1]] = [players[currIndex - 1], players[currIndex]];
        position[players[currIndex]] = currIndex;
        position[calling] = currIndex - 1;
    })
    return players;
}